subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/FileWatcherRegistry.java

```diff
@@ -31,6 +31,8 @@
 
 public interface FileWatcherRegistry extends Closeable {
 
+    boolean isWatchingAnyLocations();
```

Comment:
is there a reason this method is on two interfaces?

Reply:
Yeah, since I need to do some delegation between the different parts. There is the `BuildLifecycleAwareVirtualFileSystem`, which is a service and which holds the registry and manages its lifecycle. And that exposes the new interface as a service.

Do you have a suggestion how to make this better?

Reply:
Oh, I see. I thought that maybe this was lost during some code changes. I don't have a suggestion, I think is ok that way.